src/detect-content.c

```diff
@@ -407,21 +407,29 @@ bool DetectContentPMATCHValidateCallback(const Signature *s)
         if (sm->type != DETECT_CONTENT)
             continue;
         const DetectContentData *cd = (const DetectContentData *)sm->ctx;
-        uint32_t right_edge = cd->content_len + cd->offset;
         if (cd->content_len > max_right_edge) {
             SCLogError(SC_ERR_INVALID_SIGNATURE,
                     "signature can't match as content length %u is bigger than dsize %u.",
                     cd->content_len, max_right_edge);
-            return FALSE;
+            return false;
+        }
+
+        /* Ensure that distance values are included in validation */
+        uint32_t right_edge = cd->content_len + cd->offset;
+        if (cd->distance >= 0) {
+            right_edge += cd->distance;
+        } else {
+            right_edge = (uint32_t)abs(cd->distance) - right_edge;
```

Comment:
distance describes a relation between 2 patterns, so we'd have to look at something like
`right_edge = prev_cd->content_len + prev_cd->offset + cur_cd->distance`